control-plane/stor-port/src/types/v0/store/snapshots/volume.rs

```diff
@@ -218,16 +218,18 @@ impl SpecTransaction<VolumeSnapshotOperation> for VolumeSnapshot {
                 self.status = SpecStatus::Deleted;
             }
             VolumeSnapshotOperation::Create(info) => {
-                if let Some(result) = info.complete.lock().unwrap().take() {
+                if let Some(result) = info.complete.lock().unwrap().as_ref() {
                     self.metadata.size = result.replica.meta().size();
                     // replace-in-place the logged replica specs.
                     self.metadata
                         .transactions
-                        .insert(info.txn_id, vec![result.replica]);
+                        .insert(info.txn_id, vec![result.replica.clone()]);
                     self.status = SpecStatus::Created(());
+                } else {
+                    // means we've restarted with the op in progress... and the snapshot was not
+                    // successful!
+                    tracing::error!(?self, "Snapshot Create completion without the result");
```

Comment:
How do we reach here? Should we not bail out in the operations itself if the `VolumeSnapshotCompleter` is empty/invalid.

Reply:
We may reach there due to a bug (the one you saw before!) or if we call commit_op after container restart for whatever reason.